<a href="https://colab.research.google.com/github/STR4NG3R/AnalisisDatos/blob/main/1U/2/pymd.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Titulo: Explorar Datos jugadores Fifa
#Alumno: Pablo Eduardo Martinez Solis
#Fecha 02/09/2022

#Objetivo
Analisar, explorar datos, para recaudar variables de interes, los cuales nos permiten tener una mejor compresion de los jugadores segun sus estadisticas

In [2]:
import pandas as pd 
import matplotlib.pyplot as plt
data = pd.read_csv("https://raw.githubusercontent.com/rpizarrog/Analisis-Inteligente-de-datos/main/datos/datos.FIFA.limpios.csv", encoding= 'unicode_escape')

Describir los datos a explorar los primeros 10 registros


In [3]:
data.head(10)

Unnamed: 0               Name  Age Nationality  Overall  Potential  \
0           1           L. Messi   31   Argentina       94         94   
1           2  Cristiano Ronaldo   33    Portugal       94         94   
2           3          Neymar Jr   26      Brazil       92         93   
3           4             De Gea   27       Spain       91         93   
4           5       K. De Bruyne   27     Belgium       91         92   
5           6          E. Hazard   27     Belgium       91         91   
6           7          L. Modric   32     Croatia       91         91   
7           8          L. Suárez   31     Uruguay       91         91   
8           9       Sergio Ramos   32       Spain       91         91   
9          10           J. Oblak   25    Slovenia       90         93   

                  Club Preferred.Foot  International.Reputation  Weak.Foot  \
0         FC Barcelona           Left                       5.0        4.0   
1             Juventus          Right                       5.0        4.0   
2  Paris Saint-Germain          Right                       5.0        5.0   
3    Manchester United          Right                       4.0        3.0   
4      Manchester City          Right                       4.0        5.0   
5              Chelsea          Right                       4.0        4.0   
6          Real Madrid          Right                       4.0        4.0   
7         FC Barcelona          Right                       5.0        4.0   
8          Real Madrid          Right                       4.0        3.0   
9      Atlético Madrid          Right                       3.0        3.0   

   ...  StandingTackle SlidingTackle GKDiving  GKHandling  GKKicking  \
0  ...            28.0          26.0      6.0        11.0       15.0   
1  ...            31.0          23.0      7.0        11.0       15.0   
2  ...            24.0          33.0      9.0         9.0       15.0   
3  ...            21.0          13.0     90.0        85.0       87.0   
4  ...            58.0          51.0     15.0        13.0        5.0   
5  ...            27.0          22.0     11.0        12.0        6.0   
6  ...            76.0          73.0     13.0         9.0        7.0   
7  ...            45.0          38.0     27.0        25.0       31.0   
8  ...            92.0          91.0     11.0         8.0        9.0   
9  ...            12.0          18.0     86.0        92.0       78.0   

   GKPositioning  GKReflexes      Valor  Estatura  PesoKgs  
0           14.0         8.0  110500000      1.70    72.12  
1           14.0        11.0   77000000      1.88    83.01  
2           15.0        11.0  118500000      1.75    68.04  
3           88.0        94.0   72000000      1.93    76.20  
4           10.0        13.0  102000000      1.80    69.85  
5            8.0         8.0   93000000      1.73    73.94  
6           14.0         9.0   67000000      1.73    66.22  
7           33.0        37.0   80000000      1.83    86.18  
8            7.0        11.0   51000000      1.83    82.10  
9           88.0        89.0   68000000      1.88    87.09  

[10 rows x 50 columns]

In [4]:
data.describe()

Unnamed: 0           Age       Overall     Potential  \
count  17955.000000  17955.000000  17955.000000  17955.000000   
mean    8978.000000     25.095349     66.228237     71.323197   
std     5183.306377      4.659778      6.921015      6.139378   
min        1.000000     16.000000     46.000000     48.000000   
25%     4489.500000     21.000000     62.000000     67.000000   
50%     8978.000000     25.000000     66.000000     71.000000   
75%    13466.500000     28.000000     71.000000     75.000000   
max    17955.000000     45.000000     94.000000     95.000000   

       International.Reputation     Weak.Foot   Skill.Moves      Crossing  \
count              17907.000000  17907.000000  17907.000000  17907.000000   
mean                   1.113699      2.947339      2.363042     49.751885   
std                    0.395193      0.660232      0.756256     18.351445   
min                    1.000000      1.000000      1.000000      5.000000   
25%                    1.000000      3.000000      2.000000     38.000000   
50%                    1.000000      3.000000      2.000000     54.000000   
75%                    1.000000      3.000000      3.000000     64.000000   
max                    5.000000      5.000000      5.000000     93.000000   

          Finishing  HeadingAccuracy  ...  StandingTackle  SlidingTackle  \
count  17907.000000     17907.000000  ...    17907.000000   17907.000000   
mean      45.586698        52.298878  ...       47.684202      45.644441   
std       19.511503        17.361599  ...       21.647299      21.271215   
min        2.000000         4.000000  ...        2.000000       3.000000   
25%       30.000000        44.000000  ...       27.000000      24.000000   
50%       49.000000        56.000000  ...       55.000000      52.000000   
75%       62.000000        64.000000  ...       66.000000      64.000000   
max       95.000000        94.000000  ...       93.000000      91.000000   

           GKDiving    GKHandling     GKKicking  GKPositioning    GKReflexes  \
count  17907.000000  17907.000000  17907.000000   17907.000000  17907.000000   
mean      16.587536     16.365276     16.203105      16.356062     16.682080   
std       17.662332     16.875398     16.467295      16.989954     17.917073   
min        1.000000      1.000000      1.000000       1.000000      1.000000   
25%        8.000000      8.000000      8.000000       8.000000      8.000000   
50%       11.000000     11.000000     11.000000      11.000000     11.000000   
75%       14.000000     14.000000     14.000000      14.000000     14.000000   
max       90.000000     92.000000     91.000000      90.000000     94.000000   

              Valor      Estatura       PesoKgs  
count  1.795500e+04  17907.000000  17907.000000  
mean   2.444530e+06      1.812325     75.279478  
std    5.626715e+06      0.067671      7.077378  
min    1.000000e+04      1.550000     49.900000  
25%    3.250000e+05      1.750000     69.850000  
50%    7.000000e+05      1.800000     74.840000  
75%    2.100000e+06      1.850000     79.830000  
max    1.185000e+08      2.060000    110.220000  

[8 rows x 44 columns]

Top 5 juadores peor pagados

In [5]:
data.sort_values(by=['Valor'], ascending=True).head(5)

Unnamed: 0         Name  Age Nationality  Overall  Potential  \
16329       16330  K. Kitamoto   36       Japan       57         57   
15203       15204     M. Tyler   41     England       59         59   
15284       15285   S. Shearer   37    Scotland       59         59   
16965       16966   Y. Nishibe   37       Japan       55         55   
16077       16078     C. Weale   36     England       57         57   

                      Club Preferred.Foot  International.Reputation  \
16329          Vissel Kobe          Right                       1.0   
15203  Peterborough United          Right                       1.0   
15284        Oxford United           Left                       1.0   
16965      Shimizu S-Pulse          Right                       1.0   
16077          Exeter City           Left                       1.0   

       Weak.Foot  ...  StandingTackle SlidingTackle GKDiving  GKHandling  \
16329        2.0  ...            56.0          55.0      7.0        11.0   
15203        3.0  ...            16.0          17.0     61.0        54.0   
15284        3.0  ...            11.0          11.0     59.0        56.0   
16965        2.0  ...            12.0          12.0     54.0        59.0   
16077        2.0  ...            17.0          18.0     57.0        56.0   

       GKKicking  GKPositioning  GKReflexes  Valor  Estatura  PesoKgs  
16329        8.0            8.0        12.0  10000      1.80    78.02  
15203       54.0           63.0        55.0  10000      1.83    83.01  
15284       52.0           60.0        60.0  10000      1.91    88.90  
16965       56.0           59.0        56.0  10000      1.88    78.93  
16077       57.0           56.0        56.0  10000      1.88    83.91  

[5 rows x 50 columns]

Conteo de Jugadores que prefieren el pie para patear

In [6]:
data.groupby(['Preferred.Foot'])['Preferred.Foot'].count()

Preferred.Foot
Left      4159
Right    13748
Name: Preferred.Foot, dtype: int64

Nacion con jugadores con menor valor

In [10]:
data.groupby('Nationality')['Valor'].mean().sort_values(ascending=True).head(1)

Nationality
Puerto Rico    110000.0
Name: Valor, dtype: float64

#Interpretacion
Podemos observar que gracias a la coleccion de datos podemos apreciar que la mayoria de jugadores vienen de England, se puede apreciar que la estatura promedio de los jugadores es de 1.81, las edades promedios son de 25, hay mas jugadores derechos que zurdos, se puede observar que la nacion con jugadores con menor valor es Puerto Rico.

Se puede apreciar que en el top 5 de jugadores con mejor valor la mayoria son derechos,
a su vez japon e inglaterra comparten las naciones con mas jugadores en el top 5.

Esto nos puedo ayudar a comprender que datos son los que funcionan con los jugadore en la coleccion, esto le permite a directivos de futbol el tener un estandar de jugador.

